libfreerdp/core/smartcardlogon.c

```diff
@@ -603,6 +603,16 @@ static BOOL smartcard_hw_enumerateCerts(const rdpSettings* settings, LPCWSTR csp
 		/* load a unique CSP by pkcs11 module path */
 		LPCSTR paths[] = { Pkcs11Module, NULL };
 
+		if (!csp)
+		{
+			csp = _wcsdup(MS_SCARD_PROV);
```

Comment:
@hardening memory leak?

Reply:
Ok fixed, I thought that it would make the callees segfault but in fact they all do copies which is fine.